From 737e7e318a7c757a33c7d8bf13e82c8dc6a84b5a Mon Sep 17 00:00:00 2001
From: Igor Pecovnik <igor@armbian.com>
Date: Tue, 13 Oct 2026 21:07:18 +0200
Subject: [PATCH 14/15] boot: image-fdt: do not report a region that is
 reserved already

The board DT lists the boot ROM region in memreserve and in
reserved-memory. Do not print an error for the second entry.

Signed-off-by: Igor Pecovnik <igor@armbian.com>
---
 boot/image-fdt.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/boot/image-fdt.c b/boot/image-fdt.c
index 956a3d97c4..3b429c3036 100644
--- a/boot/image-fdt.c
+++ b/boot/image-fdt.c
@@ -91,7 +91,8 @@ static void boot_fdt_handle_region(u64 addr, u64 size, u32 flags, bool free)
 		ret = lmb_alloc_mem(LMB_MEM_ALLOC_ADDR, 0, &rsv_addr, size,
 				    flags);
 
-	if (!ret || ret == -EFAULT) {
+	/* -EEXIST: the region is reserved already, e.g. listed twice */
+	if (!ret || ret == -EFAULT || ret == -EEXIST) {
 		debug("   %s fdt memory region%s: addr=%llx size=%llx flags=%x ret=%ld\n",
 		      free ? "free" : "reserve", ret ? " failed" : "",
 		      (unsigned long long)addr, (unsigned long long)size, flags, ret);
-- 
2.53.0

